core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -2625,7 +2632,21 @@ class KafkaController(val config: KafkaConfig,
           info(s"Skipping replica leader election ($electionType) for partition $p by $electionTrigger since it doesn't exist.")
         }
 
-        val (partitionsBeingDeleted, livePartitions) = knownPartitions.partition(partition =>
+        val (knownPartitionsWithValidRequest, knownPartitionsWithInvalidRequest) = if (electionType == ElectionType.RECOMMENDED) {
+          partitionRecommendedLeadersFromAdminClientOpt match {
+            case Some(partitionRecommendedLeaders) => knownPartitions.partition(partitionRecommendedLeaders.contains(_))
+            case None => (Set.empty[TopicPartition], knownPartitions) // all known partitions are invalid
+          }
+        } else {
+          (knownPartitions, Set.empty)
+        }
+
+        if (knownPartitionsWithInvalidRequest.nonEmpty) {
+          warn(s"Skipping replica leader election ($electionType) for partitions $knownPartitionsWithInvalidRequest " +
+            s"by $electionTrigger since there is no recommended leader")
+        }
```

Comment:
I'm currently confused by this, but hopefully it's just because I haven't read far enough to find the logic I'm expected. Ergo, possible placeholder comment...

But what I'd _expect_ is that we'd fall back to whatever the normal leader-election strategy is for any partitions that have a bad recommendation. This implies we're simply ignoring those guys.

Reply:
Discussed offline; failures are expected to be super-rare, and broker-initiated leader election is rare on top of that. The fallback is the original, manually triggered approach, at least for now. Can iterate after the follow-on PRs, i.e., when v1 of the feature is finished.